src/Pgvector/Vector.cs

```diff
@@ -19,4 +19,34 @@ public override string ToString()
 
     public float[] ToArray()
         => vec;
+
+    public override bool Equals(object? obj)
+    {
+        if (obj is not Vector vector)
+        {
+            return false;
+        }
+
+        var arrayA = vec;
+        var arrayB = vector.ToArray();
```

Comment:
An Equals implementation of Vector should not be calling ToArray, which allocates and copies the entire array (that's very inefficient).

/cc @ankane

Reply:
`ToArray` here is not actually allocate a new array. It just returns the private property.

```
private float[] ToArray() 
    => vec;
```